python_modules/libraries/dagster-components/dagster_components/core/component_rendering.py

```diff
@@ -0,0 +1,97 @@
+import json
+from typing import AbstractSet, Any, Mapping, Optional, Sequence, Type, TypeVar, Union
+
+import dagster._check as check
+from dagster._record import record
+from jinja2 import Template
+from pydantic import BaseModel
+from pydantic.fields import FieldInfo
+
+T = TypeVar("T")
+
+REF_BASE = "#/$defs/"
+REF_TEMPLATE = f"{REF_BASE}{{model}}"
+
+CONTEXT_KEY = "required_rendering_context"
+
+
+def add_required_rendering_context(field: FieldInfo, context: AbstractSet[str]) -> Any:
```

Comment:
We should do some bikeshedding here as names tender on sticking around. 

Maybe `render_scope` or `rendering_scope` or `scoped_field`?

Right now upstack it looks like:

```python
class DbtProjectParams(BaseModel):
    dbt: DbtCliResource
    op: Optional[OpSpecBaseModel] = None
    translator: Optional[DbtNodeTranslatorParams] = add_required_rendering_context(
        Field(default=None), {"node"}
    )
    asset_attributes: Optional[Sequence[AssetSpecProcessorModel]] = None
```


```python
class DbtProjectParams(BaseModel):
    dbt: DbtCliResource
    op: Optional[OpSpecBaseModel] = None
    translator: Optional[DbtNodeTranslatorParams] = rendering_scope(
        Field(default=None), {"node"}
    )
    asset_attributes: Optional[Sequence[AssetSpecProcessorModel]] = None
```

Also did you consider any alternative syntaxes? Could we just `typing.Annotated` for this?

Reply:
resolved this upstack